src/rgw/rgw_op.cc

```diff
@@ -5282,34 +5287,48 @@ void RGWRestoreObj::execute(optional_yield y)
   int op_ret = s->object->get_obj_attrs(y, this);
   if (op_ret < 0) {
     ldpp_dout(this, 1) << "failed to fetch get_obj_attrs op ret = " << op_ret << dendl;
+    restore_op = op_ret;
     return;
   }
-  rgw::sal::Attrs attrs = s->object->get_attrs();
-  auto attr_iter = attrs.find(RGW_ATTR_MANIFEST);
+  attrs = s->object->get_attrs();
+  auto attr_iter = attrs.find(RGW_ATTR_RESTORE_STATUS);
```

Comment:
you do not need these checks now right..

Reply:
Not for First request ?

Reply:
As per the new changes made in handle_cloudtier_obj(), 
op_ret = 0 means first request/restore request sent
op_ret =  1 -- IN_PROGRESS
op_ret = 2 -- CLOUD_RESTORED..

if there is any issue with those return status, please fix the same.